avail-test.html

```diff
@@ -21,26 +21,42 @@
       this.vm = $scope;
 
       console.log("Starting Avail Tester");
-      var baseTime = moment().hour(13).minutes(0); // start a 13:00 today
+      var baseTime = moment().hour(1).minutes(0).seconds(0); // start at 1 AM today
 
       $scope.$on('AvailChartTimeRangeChanged', function (event, data) {
         console.info('Received AvailChartTimeRangeChanged: '+data[0] +' - '+ data[1]);
+        $scope.startTime = data[0];
+        $scope.endTime = data[1];
+        $scope.refreshChartWithDateRange();
       });
 
-      this.availAllChartData = [
-        {'timestamp': +baseTime.add('minutes', 5), 'value': 'up'},
-        {'timestamp': +baseTime.add('minutes', 10), 'value': 'down'},
-        {'timestamp': +baseTime.add('minutes', 10), 'value': 'up'},
-        {'timestamp': +baseTime.add('minutes', 20), 'value': 'unknown'},
-        {'timestamp': +baseTime.add('minutes', 15), 'value': 'down'}];
+      $scope.refreshChartWithDateRange = function () {
+        console.info('refreshChartWithDateRange');
+        // we already have the data, but perhaps the data is stale and you want to requery
+        // also, by just changing the data the charts will automatically update themselves
+        $scope.availAllChartData = $scope.availAllChartData.filter(function (value) {
+          return new Date(value.timestamp) >= $scope.startTime && new Date(value.timestamp) <= $scope.endTime;
+        });
+        $scope.$digest();
 
-      this.startTime = baseTime;
-      this.endTime = this.availAllChartData[this.availAllChartData.length - 1].timestamp;
+      };
 
-      console.log("StartTime: " + moment(baseTime).toISOString());
-      console.log("EndTime: " + moment().toISOString());
+      $scope.availAllChartData = [
```

Comment:
The "this" notation wasn't actually working (guess because of $scope.$on(...) scope) and so I made it consistent/according with what we also have in the main examples page, https://github.com/hawkular/hawkular-charts/blob/master/index.html#L96 . But I agree this should be improved globally, I can revisit this after.